packages/nostr/src/client/index.ts

```diff
@@ -18,62 +12,53 @@ export class Nostr {
   /**
    * Open connections to relays.
    */
-  readonly #conns: Map<
-    string,
-    {
-      conn: Conn
-      /**
-       * Has this connection been authenticated via NIP-44 AUTH?
-       */
-      auth: boolean
-      /**
-       * Should this connection be used for receiving messages?
-       */
-      read: boolean
-      /**
-       * Should this connection be used for publishing events?
-       */
-      write: boolean
-    }
-  >
+  readonly #conns: Map<string, ConnState> = new Map()
 
   /**
    * Mapping of subscription IDs to corresponding filters.
    */
-  readonly #subscriptions: Map<string, Filters[]> = new Map()
+  readonly #subscriptions: Map<string, Filters> = new Map()
 
-  readonly #eventCallbacks: EventCallback[] = []
-  readonly #noticeCallbacks: NoticeCallback[] = []
-  readonly #errorCallbacks: ErrorCallback[] = []
+  #eventCallback?: EventCallback
+  #noticeCallback?: NoticeCallback
+  #errorCallback?: ErrorCallback
 
   /**
    * Add a new callback for received events.
    */
-  onEvent(cb: EventCallback): void {
-    this.#eventCallbacks.push(cb)
-  }
-
+  on(on: "event", cb: EventCallback | undefined | null): void
   /**
    * Add a new callback for received notices.
    */
-  onNotice(cb: NoticeCallback): void {
-    this.#noticeCallbacks.push(cb)
-  }
-
+  on(on: "notice", cb: NoticeCallback | undefined | null): void
   /**
    * Add a new callback for errors.
    */
-  onError(cb: ErrorCallback): void {
-    this.#errorCallbacks.push(cb)
+  on(on: "error", cb: ErrorCallback | undefined | null): void
+  // TODO Also add on: ("subscribed", subscriptionId) which checks "OK"/"NOTICE" and makes a callback?
+  // TODO Also add on: ("sent", eventId) which checks "OK"/"NOTICE" and makes a callback?
```

Comment:
Can be added later, we dont currently use OK callback on snort but it would be nice to get confirmation when trying to publish events